chan_handler.go

```diff
@@ -20,10 +22,11 @@ type StateChange struct {
 	To State
 }
 
-func NewChanHandler(scCh chan<- StateChange, errCh chan<- error) *ChanHandler {
+func NewChanHandler(logHandler LogPositionHandler, scCh chan<- StateChange, errCh chan<- error) *ChanHandler {
```

Comment:
If possible and not too awkward, that would be best indeed. If not using new API, old behavior. Internally, it means that we would not invoke the callback.